usersession/userd/launcher.go

```diff
@@ -123,6 +130,126 @@ func (s *Launcher) OpenURL(addr string, sender dbus.Sender) *dbus.Error {
 	return nil
 }
 
+// OpenDesktopEntryEnv implements the 'OpenDesktopEntryEnv' method of the 'io.snapcraft.Launcher'
+// DBus interface.
+func (s *Launcher) OpenDesktopEntryEnv(desktop_file_id string, env []string, sender dbus.Sender) *dbus.Error {
+	desktop_file, err := desktopFileIdToFilename(desktop_file_id)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	exec_command, err := readExecCommandFromDesktopFile(desktop_file)
+
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	args, err := shlex.Split(exec_command)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	// Passing exec variables between confined snaps raises unanswered questions and they are not required
+	// for the simple cases.  For now, we don't have support for passing them in the dbus API and drop
+	// them from the command.
+	// https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables
+	i := 0
```

Comment:
I would mildly prefer if we error on those explicitly. It's better to say "this is not supported now, it may be supported later" than change semantics between two versions without letting the other side know what happened. Is this a reasonable approach or would this rule out too many applications?

Reply:
The "other side" in this case is the process invoking the dbus method. Until we add a method that supports passing exec variables, the "other side" has no way to affect the behaviour.

To quote the [desktop-entry-spec](https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables):
>  Deprecated field codes should be removed from the command line and ignored. 
...
>  If the application should not open any file the %f, %u, %F and %U field codes must be removed from the command line and ignored. 

We _could_ discriminate amongst the possible codes that appear in the .desktop entry, and only drop those that can be omitted (the above plus "%i" and assorted deprecated codes) and error on those that are invalid or we don't support (e.g. "%k"). But I don't think that would be useful.

Reply:
I was not familiar with the spec. I should have read the reference, sorry.